### PR TITLE
Update streamly

### PR DESCRIPTION
**Pull request by adithyaov**

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -49,22 +49,6 @@ let haskellPackages =
                         mkPackage super "streamly-examples"
                             ./. flags inShell;
 
-                    streamly =
```
Comment (harendra-kumar) [inline default.nix:52]: Why remove this? We will need to add this again for 0.8.1 when we release.

File: cabal.project [hunk 2]
```diff
@@ -1,3 +1,13 @@
...
 
+source-repository-package
+    type:       git
+    location:   https://github.com/composewell/streamly
+    tag:        f1a04362162d707054115a34563e1c35199bb796
+
+source-repository-package
+    type:       git
+    location:   https://github.com/composewell/unicode-data
+    tag:        e3208a437b4b71613656c0aff9165442256f89ed
```
Comment (harendra-kumar) [inline cabal.project:11]: We can add the released package to default.nix, instead of adding this here.

Comment (harendra-kumar) [reply cabal.project:11]: Like this: https://github.com/composewell/streamly-examples/commit/0926cd29365671ce91c1dd3bce21eb52b320c194

File: examples/WordCountParallelUTF8.hs [hunk 3]
```diff
@@ -179,16 +179,16 @@ data Field =
...
-writeField v fld = void . MArray.unsafeWriteIndex v (fromEnum fld)
+writeField v fld = void . MArray.putIndexUnsafe v (fromEnum fld)
 
 modifyField :: MArray.Array Int -> Field -> (Int -> Int) -> IO ()
 modifyField v fld f = do
   let index = fromEnum fld
-  val <- MArray.unsafeIndexIO v index
-  void $ MArray.unsafeWriteIndex v index (f val)
+  val <- MArray.getIndexUnsafe v index
+  void $ MArray.putIndexUnsafe v index (f val)
```
Comment (harendra-kumar) [inline examples/WordCountParallelUTF8.hs:191]: We can use `modifyIndexUnsafe` instead.

File: default.nix (hunk 1 above)
Comment (adithyaov) [reply default.nix:52]: We'll add this back when we release streamly.

File: cabal.project [hunk 4]
```diff
@@ -1,3 +1,15 @@
...
+
+source-repository-package
+    type:       git
+    location:   https://github.com/composewell/streamly
+    tag:        f1a04362162d707054115a34563e1c35199bb796
+
+source-repository-package
+    type:       git
+    location:   http://github.com/composewell/unicode-data
+    tag:        91bc99182b39290b423c4a584a0b077d91eead4c
```
Comment (harendra-kumar) [inline cabal.project:15]: unicode-data is already released why do we need a source-repository-package for this?

File: default.nix [hunk 5]
```diff
@@ -49,21 +49,29 @@ let haskellPackages =
...
+                    #           else [];
+                    #         enableLibraryProfiling = false;
+                    #         doHaddock = false;
+
+                    # unicode-data =
+                    #   super.callHackageDirect
+                    #     { pkg = "unicode-data";
+                    #       ver = "0.2.0";
+                    #       sha256 = "073wbhdxj1sh5160blaihbzkkhabs8s71pqhag16lvmgbb7a3hla";
+                    #     } {};
```
Comment (harendra-kumar) [inline default.nix:74]: Why do we need to even comment it? Won't cabal override it from the cabal.project file?

File: examples/Intro.hs [hunk 6]
```diff
@@ -49,28 +49,19 @@ sumInt1 =
 -- nested loops (or cross product) when the two streams do not depend on each
 -- other. The loops fuse completely generating code equivalent to C.
 crossProduct :: (Int,Int) -> (Int,Int) -> Identity Int
-crossProduct (from1,to1) (from2,to2) =
+crossProduct ft1 ft2 =
```
Comment (harendra-kumar) [inline examples/Intro.hs:52]: Call it `range1`, `range2`.

File: default.nix (hunk 5 above)
Comment (adithyaov) [reply default.nix:74]: I've been having some nix errors that keeps saying `unable to find unicode-data` although I add this. It does even check the `sha256`. For the time being, I don't want to debug that.
Some caching issue probably.

File: cabal.project (hunk 4 above)
Comment (adithyaov) [reply cabal.project:15]: It's for local development. Ah, I should just use `cabal.project.local`

File: default.nix (hunk 5 above)
Comment (harendra-kumar) [reply default.nix:74]: I had this working on the split branch (or may be locally let me check).

Comment (harendra-kumar) [reply default.nix:74]: Add this in additionalDeps:

```
              executableHaskellDepends = with hspkgs; [
                unicode-data
              ];
```

Comment (adithyaov) [reply default.nix:74]: I still have the same issue.

Comment (adithyaov) [reply default.nix:74]: Please ignore this. This will be reverted anyway.

Comment (harendra-kumar) [reply default.nix:74]: Maybe you have the sha256 of some other package? In that case it successfully matches the package, does not give any error and uses the wrong package instead.

File: .circleci/config.yml [hunk 7]
```diff
@@ -321,7 +321,8 @@ jobs:
            environment:
              BUILD: "cabal-v2"
              CABAL_BUILD_OPTIONS: "--flag fusion-plugin"
-             DISABLE_SDIST_BUILD: "no"
+             DISABLE_SDIST_BUILD: "yes"
+             CABAL_PROJECT: "cabal.project"
```
Comment (harendra-kumar) [inline .circleci/config.yml:325]: Are you going to commit these?

Comment (adithyaov) [reply .circleci/config.yml:325]: Yes, I'll revert them later.

Comment (adithyaov) [reply .circleci/config.yml:325]: Before releasing. Or should we keep the CIs failing?

Comment (adithyaov) [reply .circleci/config.yml:325]: Or keep this PR open till we upload streamly?

**Comment by harendra-kumar** [comment]
Looks good. Assuming CI change is temp.

**Comment by adithyaov** [comment]
See #34

**Pull request closed without merging**